cocos/scripting/js-bindings/script/jsb_boot.js

```diff
@@ -74,6 +74,13 @@ cc.view.getTargetDensityDPI = function() {return cc.macro.DENSITYDPI_DEVICE;};
 
 cc.eventManager = cc.director.getEventDispatcher();
 
+cc.EventDispatcher.prototype._addCustomListener = cc.EventDispatcher.prototype.addCustomListener;
+cc.EventDispatcher.prototype.addCustomListener = function(eventName, callback) {
+    var ret = this._addCustomListener(eventName, callback);
+    jsb.registerNativeRef(cc.eventManager, ret);
```

Comment:
addCustomListener return a EventListenerCustom which isn't retained in c++, so we need to retain it before this function returns.